src/account/PluginManager2.sol

```diff
@@ -73,19 +74,21 @@ abstract contract PluginManager2 {
             }
         }
 
-        _validationData.isGlobal = validationConfig.isGlobal();
-        _validationData.isSignatureValidation = validationConfig.isSignatureValidation();
-
         for (uint256 i = 0; i < selectors.length; ++i) {
             bytes4 selector = selectors[i];
             if (!_validationData.selectors.add(toSetValue(selector))) {
                 revert ValidationAlreadySet(selector, validationConfig.functionReference());
             }
         }
 
-        if (installData.length > 0) {
-            address plugin = validationConfig.plugin();
-            IPlugin(plugin).onInstall(installData);
+        if (validationConfig.functionId() != _SELF_PERMIT_VALIDATION_FUNCTIONID) {
+            // Only allow global validations and signature validations if they're not direct-call validations.
```

Comment:
Is this necessary to prevent an access control issue? Or is it just an anti-footgun measure?

Reply:
It's just an anti-footgun, though I've not evaluated the consequences of having global validation for a direct call-- signature validation either.

Reply:
Ok sounds good. It might help to support global/sig validation in some very niche cases where the direct call validation is used to create an "owner" EOA, but that seems small enough to ignore for now.

Reply:
Gotcha, happy to revisit this down the line if we see the feature's important!